### ISSUE TITLE
Create path if it doesn't exist?

### ISSUE THREAD

**Issue by jptaylor** [substantial/updeep#20]
It would be good if `updateIn` could function more like lodash `_.set`

```
const obj = { foo: null }
const obj2 = u.updateIn('foo.bar', 1, obj);
// {foo: {bar:1}}
```
Perhaps make this optional?

Thanks for the great library.

**Comment by aaronjensen** [comment]
Thanks, you're welcome!

I'd be good w/ this. Should it also create an array if a number is used?

```js
const obj = { foo: null }
const obj2 = u.updateIn('foo.0', 1, obj);
// {foo: [1]}
```

**Comment by mindjuice** [comment]
With array creation, if you create the array, you'll be able to create arrays with holes.  This is fine if you know what you're doing though, so it would just need to be documented.

Also, I assume object/array creation would be done as necessary at each part of the path.  So in the example, ``const obj = {};`` would also work.

Hmmmm...what about ``var foo = undefined;``?  Would it create the initial object?

**Comment by aaronjensen** [comment]
Ok, so, I looked at this a little bit. If you pass `{}` instead of `{ foo: null }` it works as you are wanting. (@mindjuice it also works w/ undefined as the root object already, all the update functions default to an object if its undefined)

What doesn't work is when the value is null. When it's null it actually throws an error, so that seems like a bug to me. I'm not entirely sure how to handle it, but I'm leaning towards defaulting it the same way that I do `undefined`.

**Comment by mindjuice** [comment]
I would say that if your path starts with a number, then you should create an array instead of an object if the object is ``undefined``.

    const arr = u.updateIn('3.foo', 1, undefined);
    // [,,,{foo: 1}]

On the one hand, the case for ``null`` seems like it should behave the same way.  What if the value is some other type like number or string?

**Comment by mindjuice** [comment]
For ``null``, I'm just trying to think if I would ever want to **not** update if the object shouldn't exist.  I can always explicitly check for that case if it ever occurs I suppose.

I would either have ``null`` behave the same as ``undefined`` or perhaps just do nothing and return ``null``.

**Comment by mindjuice** [comment]
Using a number for the path and passing in a string could actually make sense:

    const str = 'abc';
    const newStr = u.updateIn([1], 'B', str);
    // "aBc"

Although a bit of a weird way to do it instead of using normal string functions. :-)

Just a thought.  Probably not worth it.

**Comment by jptaylor** [comment]
@aaronjensen For my use case, `null` would preferably act the same as `undefined`. I'm struggling to think up a scenario where this wouldn't be desirable?

**Comment by aaronjensen** [comment]
Ok, that sounds good to me @jptaylor, want to submit a failing test or a fix?

**Comment by aaronjensen** [comment]
Fixed in 0.7.0

Array idea extracted to https://github.com/substantial/updeep/issues/22

**Comment by jptaylor** [comment]
@aaronjensen sorry, manic day yesterday. Works like a charm, thank you.
